internal/clients/team1/client.go

```diff
@@ -2,23 +2,22 @@
 package team1
 
 import (
-	"fmt"
-	"log"
-
 	"github.com/SOMAS2020/SOMAS2020/internal/common"
 	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
 )
 
 const id = shared.Team1
 
 func init() {
-	common.RegisterClient(id, &client{id: id})
+	common.RegisterClient(id, &client{id: id, Client: common.NewClient()})

```

Comment:
https://github.com/SOMAS2020/SOMAS2020/blob/56a3768df2793e44a283eba5ef6aec03c7603814/internal/server/clientreg_test.go#L12

Here we have a test to make sure all 6 clients are registered into `common.RegisteredClients`. I reckon here is a good place to strengthen that test and make sure `common.Client != nil` for all clients :)

Reply:
something like this: 
```
func TestClientReg(t *testing.T) {
	for _, client := range common.RegisteredClients {
		if client == nil {
			t.Errorf("Client '%v' does not have a Client implementation", client.GetID())
		}
	}
}
```

Reply:
whoops! nope, actually, you need `if client.Client == nil`.